### PR TITLE
Updated Zammad

### PR DESCRIPTION
**Pull request by monotek**

### CODE REVIEW

**Comment by cedvan** [comment]
:+1:

File: templates/zammad/1/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    depends_on:
+      - zammad
+    image: zammad/zammad-docker-compose:zammad-nginx
+    labels:
+      io.rancher.container.pull_image: always
+    links:
+      - zammad-railsserver
+      - zammad-websocket
+    ports:
+      - "80:80"
```
Comment (cedvan) [inline templates/zammad/1/docker-compose.yml]: Not necessary, it's load balancer responsability ;)

Comment (monotek) [reply templates/zammad/1/docker-compose.yml]: its removed now.

Comment (rawmind0) [reply templates/zammad/1/docker-compose.yml]: Just removed at version 1.5, `templates/zammad/3/docker-compose.yml`, but not added load balancer. No service published.

**Comment by rawmind0** [comment]
You are using same generic docker image tags in all package versions. Please, don't use generic docker image tags, they are not reproducible, use concrete docker image tags, corresponding to each package version. Also, please remove `io.rancher.container.pull_image: always` labels.

**Comment by monotek** [comment]
@rawmind0 
Can you help me out with the failed build?
Whats wrong?

**Comment by rawmind0** [comment]
Your `templates/zammad/3/rancher-compose.yml` file is wrong. You are mixing version 1 and 2 notation.

File: templates/zammad/3/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      description: |
+        Set true to avoid vm.max_map_count errors.
+        WARN: If set true, host param vm.max_map_count will be update to 262144.
+      default: false
+      required: true
+      type: enum
+      options:
+        - false
+        - true
+
```
Comment (rawmind0) [inline templates/zammad/3/rancher-compose.yml]: All of these are services....
```
+services:
```

File: templates/zammad/3/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    zammad-railsserver:
+      scale: 1
+      start_on_create: true
+
+    zammad-websocket:
+      scale: 1
+      start_on_create: true
+
+  services:
```
Comment (rawmind0) [inline templates/zammad/3/rancher-compose.yml]: Services has to be opened before....
```
-  services:
```

File: templates/zammad/3/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
+.catalog:
```
Comment (rawmind0) [inline templates/zammad/3/rancher-compose.yml]: Version 2 notation...
```
-.catalog:
+version: '2'
+catalog:
```

File: templates/zammad/3/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    zammad-lb:
+      scale: 1
+      lb_config:
+        port_rules:
+        - priority: 1
+          protocol: http
+          source_port: 9797
+          target_port: 80
+          service: zammad-nginx
```
Comment (rawmind0) [inline templates/zammad/3/rancher-compose.yml]: All services have bad indent. Indent one to the left.

**Comment by monotek** [comment]
Thanks a lot. Its fixed now.

**Comment by rawmind0** [comment]
One more thing @monotek . Could you please remove `templates/zammad/1` and `templates/zammad/2` folder and just add `templates/zammad/3` renamed to `templates/zammad/1`??

**Comment by monotek** [comment]
Done :-)

File: templates/zammad/1/docker-compose.yml.tpl [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      - zammad-postgresql
+    restart: always
+    volumes:
+      - zammad-backup:/var/tmp/zammad
+      - zammad-data:/opt/zammad
+
+  zammad-elasticsearch:
+    image: zammad/zammad-docker-compose:zammad-elasticsearch-2.2.0-12
+    labels:
+      io.rancher.sidekicks: {{- if eq .Values.UPDATE_SYSCTL "true" -}}zammad-es-sysctl{{- end}}
```
Comment (rawmind0) [inline templates/zammad/1/docker-compose.yml.tpl]: Labels definition is wrong. If command has to include full labels definition.
```
-    labels:
-      io.rancher.sidekicks: {{- if eq .Values.UPDATE_SYSCTL "true" -}}zammad-es-sysctl{{- end}}
+    {{- if eq .Values.UPDATE_SYSCTL "true" }}
+    labels:
+      io.rancher.sidekicks: zammad-es-sysctl
+    {{- end}}
```

File: templates/zammad/1/docker-compose.yml.tpl [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+  {{- end}}
+
+  zammad-lb:
+    image: rancher/lb-service-haproxy:v0.7.15
+    ports:
+      - 9797:9797/tcp
+    labels:
+      io.rancher.container.agent.role: environmentAdmin,agent
+      io.rancher.container.agent_service.drain_provider: 'true'
+      io.rancher.container.create_agent: 'true'
```
Comment (rawmind0) [inline templates/zammad/1/docker-compose.yml.tpl]: Don't need to define labels here. They are implicit. 
```
-    labels:
-      io.rancher.container.agent.role: environmentAdmin,agent
-      io.rancher.container.agent_service.drain_provider: 'true'
-      io.rancher.container.create_agent: 'true'
```

File: templates/zammad/1/docker-compose.yml.tpl [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    network_mode: none
+    image: rawmind/alpine-sysctl:0.1
+    privileged: true
+    environment:
+        - "SYSCTL_KEY=vm.max_map_count"
+        - "SYSCTL_VALUE=262144"
+  {{- end}}
+
+  zammad-lb:
+    image: rancher/lb-service-haproxy:v0.7.15
```
Comment (rawmind0) [inline templates/zammad/1/docker-compose.yml.tpl]: Lb service tag is wrong, current one in v0.7.9
```
-    image: rancher/lb-service-haproxy:v0.7.15
+    image: rancher/lb-service-haproxy:v0.7.9
```

File: templates/zammad/1/rancher-compose.yml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    start_on_create: true
+
+  zammad-websocket:
+    scale: 1
+    start_on_create: true
+
+  zammad-nginx:
+    scale: 1
+    start_on_create: true
+    drain_timeout_ms: 10000
```
Comment (rawmind0) [inline templates/zammad/1/rancher-compose.yml]: What is this?? It's failing.
```
-    drain_timeout_ms: 10000
```

Also, what do you think to add healthcheck to zammad-nginx??
```
+    health_check:
+      healthy_threshold: 2
+      response_timeout: 2000
+      port: 80
+      unhealthy_threshold: 3
+      initializing_timeout: 300000
+      interval: 2000
+      strategy: recreate
+      request_line: GET "/" "HTTP/1.0"
```

File: templates/zammad/1/rancher-compose.yml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+      label: "Update host sysctl:"
+      description: |
+        Set true to avoid vm.max_map_count errors.
+        WARN: If set true, host param vm.max_map_count will be update to 262144.
+      default: false
+      required: true
+      type: enum
+      options:
+        - false
+        - true
```
Comment (rawmind0) [inline templates/zammad/1/rancher-compose.yml]: What do you think to add configurable published port??
```
+    - variable: "PUBLISH_PORT"
+      label: "Publish port"
+      description: |
+        Set port to publish the service.
+      required: true
+      type: "int"
+      default: 9797
```

File: templates/zammad/1/docker-compose.yml.tpl [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    ports:
+      - 9797:9797/tcp
```
Comment (rawmind0) [inline templates/zammad/1/docker-compose.yml.tpl]: What do you think to add configurable published port??
```
-      - 9797:9797/tcp
+      - ${PUBLISH_PORT}:${PUBLISH_PORT}/tcp
```

File: templates/zammad/1/rancher-compose.yml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    start_on_create: true
+    drain_timeout_ms: 10000
+
+  zammad-lb:
+    scale: 1
+    lb_config:
+      port_rules:
+      - priority: 1
+        protocol: http
+        source_port: 9797
```
Comment (rawmind0) [inline templates/zammad/1/rancher-compose.yml]: What do you think to add configurable published port??
```
-        source_port: 9797
+        source_port: ${PUBLISH_PORT}
```

**Comment by rawmind0** [comment]
Thanks for changes @monotek. Package is not working, could you please take a look at changes requested??

**Comment by monotek** [comment]
Do i need a template.yml or is it possible to add 'defaultVersion: "2.2.0-12"' to the config.yml?

File: templates/zammad/config.yml [hunk 13]
```diff
@@ -1,7 +1,8 @@
 name: Zammad
 description: |
   Zammad is a web based open source helpdesk/customer support system
-version: "1.2"
+version: "2.2.0-12"
+defaultVersion: "2.2.0-12"
```
Comment (rawmind0) [inline templates/zammad/config.yml]: `defaultVersion`parameter doesn't exist. Please remove it.
`version` parameter at this file, defines the default version.
```
-defaultVersion: "2.2.0-12"
```

File: templates/zammad/1/rancher-compose.yml [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        - false
+        - true
+
+    - variable: "PUBLISH_PORT"
+    label: "Publish port"
+    description: |
+      Set port to publish the service.
+    required: true
+    type: "int"
+    default: 9797
```
Comment (rawmind0) [inline templates/zammad/1/rancher-compose.yml]: Bad indented, please indent correctly. Just need to paste what i wrote.
```
    - variable: "PUBLISH_PORT"
-    label: "Publish port"
-    description: |
-      Set port to publish the service.
-    required: true
-    type: "int"
-    default: 9797
+      label: "Publish port"
+      description: |
+        Set port to publish the service.
+      required: true
+      type: "int"
+      default: 9797
```

**Comment by monotek** [comment]
Thanks :-) 
The drone error output was misleading for me:

```
# default version must be defined
assert template.defaultVersion
# template with default version must be defined
assert template.versionLinks[template.defaultVersion]
```

**Comment by rawmind0** [comment]
No problem. Build was failing due to indent problem.

Tested and working now. :)

LGTM...

**Comment by monotek** [comment]
Thanks for your help :-)

**Pull request merged**